hardware/national_instruments_x_series.py

```diff
@@ -165,6 +168,14 @@ def on_activate(self):
     def on_deactivate(self):
         """ Shut down the NI card.
         """
+        self._stop_analog_output()
+        # clear the task
+        try:
+            daq.DAQmxClearTask(self._scanner_ao_task)
+            self._scanner_ao_task = None
+        except:
+            self.log.exception('Could not clear AO Out Task.')
+
```

Comment:
Isn't that covered with `self.reset_hardware()`? If not, it should be.

Reply:
We thought so too, but we had the case where if we are also accessing the same hardware with another module and then call `reset`, sometimes the old tasks linger around.
So I though better save then sorry.